src/main.rs

```diff
@@ -106,3 +141,58 @@ fn print_category(
         }
     }
 }
+
+#[derive(Clone)]
+struct VersionTagParser;
+
+impl VersionTagParser {
+    fn make_error(cmd: &clap::Command, arg: Option<&clap::Arg>, value: &str) -> clap::Error {
+        let mut err = clap::Error::new(clap::error::ErrorKind::ValueValidation).with_cmd(cmd);
+        if let Some(arg) = arg {
+            err.insert(clap::error::ContextKind::InvalidArg, clap::error::ContextValue::String(arg.to_string()));
+        }
+        err.insert(clap::error::ContextKind::InvalidValue, clap::error::ContextValue::String(value.to_owned()));
+        err
+    }
+}
+
+impl clap::builder::TypedValueParser for VersionTagParser {
+
+    type Value = semver::Version;
+
+    fn parse_ref(
+        &self,
+        cmd: &clap::Command,
+        arg: Option<&clap::Arg>,
+        value: &std::ffi::OsStr,
+    ) -> Result<Self::Value, clap::Error> {
+
+        lazy_static! {
+            static ref RE: Regex = Regex::new(r"^v?(?P<version>\d+\.\d+\.\d+)$").unwrap();
+        }
+
+        let arg_value = value.to_string_lossy();
+
+        let result = RE
+            .captures(&arg_value)
+            .map(|x| x.name("version"))
+            .flatten()
+            .map(|x| x.as_str())
+            .map(|x| semver::Version::parse(x))
+            .ok_or_else(|| VersionTagParser::make_error(cmd, arg, &arg_value))
+            .map(|x| {
```

Comment:
I think this ends up a bit jank - I can take a peek at how you could maybe clean it up tn

Reply:
I’m fluent in fluent.